### ISSUE TITLE
What version of nodejs should we be using for current code

### ISSUE THREAD

**Issue by mdm001** [dvandal/cryptonote-nodejs-pool#373]
running on Ubuntu 16.04

I've followed the instructions here and am having difficulty after having it run great for a few months with the exception of not picking up the network stats correctly (especially the block height.)  I was hoping to fix this with the current code.

After I followed the git pull / npm update instructions I can't get the pool up and running again.  Then I attempted a full re-install and get the same results.

My question is what version of nodejs should I be using with the current code.  I've tried using 4.0 and 8.11.3, deleting node_modules, etc. as it states in the README but doing that yield errors like:

master_error.log:

2018-10-10 07:15:25 Pool fork 2 died, spawning replacement worker...
2018-10-10 07:15:25 API died, spawning replacement...
2018-10-10 07:15:26 Payment processor died, spawning replacement...
2018-10-10 07:15:26 Block unlocker died, spawning replacement...

/var/log/syslog:

Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]: /pool/node_modules/bignum/node_modules/bindings/bindings.js:88
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:         throw e
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:         ^
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]: Error: The module '/pool/node_modules/bignum/build/Release/bignum.node'
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]: was compiled against a different Node.js version using
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]: NODE_MODULE_VERSION 46. This version of Node.js requires
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]: NODE_MODULE_VERSION 64. Please try re-compiling or re-installing
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]: the module (for instance, using `npm rebuild` or `npm install`).
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:     at Object.Module._extensions..node (internal/modules/cjs/loader.js:718:18)
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:     at Module.load (internal/modules/cjs/loader.js:599:32)
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:     at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:     at Function.Module._load (internal/modules/cjs/loader.js:530:3)
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:     at Module.require (internal/modules/cjs/loader.js:637:17)
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:     at require (internal/modules/cjs/helpers.js:20:18)
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:     at bindings (/pool/node_modules/bignum/node_modules/bindings/bindings.js:81:44)
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:     at Object.<anonymous> (/pool/node_modules/bignum/index.js:1:92)
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:     at Module._compile (internal/modules/cjs/loader.js:689:30)
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
Oct 10 08:35:47 mmmoneropool cryptonote-nodejs-pool[12685]: #033[31m2018-10-10 08:35:47#033[39m#033[1m#033[37m [master] #033[39m#033[22mPool fork 1 died, spawning replacement worker...

Thank you for any help you can provide.

**Comment by djavdat** [comment]
Try to delete /node_modules/
```
rm -rf node_modules/
nvm install 4.0
nvm use 4.0
npm update
node init.js -config='your config.json'
```

**Comment by mdm001** [comment]
Many thanks djavdat!  Oddly, even though I had attempted what you proposed several times yesterday, I tried again a few more times today.  Although the "4.0" in your commands didn't work, "8.11.3" substituted for the "4.0" did eventually work.  One thing I wasn't doing last night was using the "f" parameter in the rm statement as I didn't see that it had helped anyone before.  Now I just have to get it to run as a service again, instead of using the node "init.js" statement.  It looks like my block height and other network stats that were off before are fixed too!
